core/vm/lightclient/types.go

```diff
@@ -228,17 +232,29 @@ func (kvmp *KeyValueMerkleProof) SetVerifiers(verifiers []merkle.ProofOpVerifier
 	kvmp.verifiers = verifiers
 }
 
+func (kvmp *KeyValueMerkleProof) SetOpsVerifier(verifier merkle.ProofOpsVerifier) {
+	kvmp.proofOpsVerifier = verifier
+}
+
+func (kvmp *KeyValueMerkleProof) SetKeyChecker(keyChecker KeyChecker) {
+	kvmp.keyChecker = keyChecker
+}
+
 func (kvmp *KeyValueMerkleProof) Validate() bool {
 	kp := merkle.KeyPath{}
 	kp = kp.AppendKey([]byte(kvmp.StoreName), merkle.KeyEncodingURL)
```

Comment:
Please do the `keyChecker` before appendkey

Reply:
sure